src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Setup/DotNetEnvironment.cs

```diff
@@ -0,0 +1,123 @@
+    }
+
+    /// <inheritdoc/>
+    public Task<bool> IsSdkInstalledAsync(string sdkVersion)
+    {
+        try
+        {
+            string archSubKey = GetArchitectureSubKey(_environment.ProcessArchitecture);
+            string registryKey = $@"SOFTWARE\dotnet\Setup\InstalledVersions\{archSubKey}\sdk";
+
+            // Get all value names from the sdk subkey
+            string[] installedVersions = _registry.GetValueNames(
+                Win32.RegistryHive.LocalMachine,
+                Win32.RegistryView.Registry32,
+                registryKey);
+
+            // Check if the requested SDK version is in the list
+            foreach (string installedVersion in installedVersions)
+            {
+                if (string.Equals(installedVersion, sdkVersion, StringComparison.OrdinalIgnoreCase))
+                {
+                    return Task.FromResult(true);
+                }
+            }
+
+            return Task.FromResult(false);
+        }
+        catch
+        {
+            // If we fail to check, assume the SDK is not installed
+            return Task.FromResult(false);
+        }
+    }
+
+    /// <inheritdoc/>
+    public string? GetDotNetHostPath()
+    {
+        // First check the registry
+        string archSubKey = GetArchitectureSubKey(_environment.ProcessArchitecture);
+        string registryKey = $@"SOFTWARE\dotnet\Setup\InstalledVersions\{archSubKey}";
+
+        string? installLocation = _registry.GetValue(
+            Win32.RegistryHive.LocalMachine,
+            Win32.RegistryView.Registry32,
+            registryKey,
+            "InstallLocation");
+
+        if (!string.IsNullOrEmpty(installLocation))
+        {
+            string dotnetExePath = Path.Combine(installLocation, "dotnet.exe");
+            if (_fileSystem.FileExists(dotnetExePath))
+            {
+                return dotnetExePath;
+            }
+        }
+
+        // Fallback to Program Files
+        string programFiles = _environment.GetFolderPath(Environment.SpecialFolder.ProgramFiles);
+        string dotnetPath = Path.Combine(programFiles, "dotnet", "dotnet.exe");
+        
+        if (_fileSystem.FileExists(dotnetPath))
+        {
+            return dotnetPath;
+        }
+
+        return null;
+    }
+
+    /// <inheritdoc/>
+    public string[]? GetInstalledRuntimeVersions(Architecture architecture)
+    {
+        // https://github.com/dotnet/designs/blob/96d2ddad13dcb795ff2c5c6a051753363bdfcf7d/accepted/2020/install-locations.md#globally-registered-install-location-new
+
+        string archSubKey = GetArchitectureSubKey(architecture);
+        string registryKey = $@"SOFTWARE\dotnet\Setup\InstalledVersions\{archSubKey}\sharedfx\Microsoft.NETCore.App";
+
+        string[] valueNames = _registry.GetValueNames(
+            Win32.RegistryHive.LocalMachine,
+            Win32.RegistryView.Registry32,
+            registryKey);
+
+        // Return null if no values found (consistent with original implementation)
```

Comment:
nit: Is the "original implementation" the code prior to this PR? If so, I don't think that comment will be clear once this merges. Asked another way, is this a "relative" comment, or an "absolute" comment? Relative (to prior code) comments are great on the diff/PR, but can be confusing in the code itself.